.github/workflows/ci.yml

```diff
@@ -27,6 +27,20 @@ jobs:
         with:
           version: "3.15.8"
 
+  test_go_oldest:
+    name: Go tests with previous Go version
+    runs-on: ubuntu-latest
+    container:
+      # The go version in this image should be N-1 wrt test_go.
+      image: quay.io/prometheus/golang-builder:1.21-base
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - run: make build
+      - run: make test
+      - run: go test ./tsdb/ -test.tsdb-isolation=false
+      - run: go test --tags=stringlabels ./...
+      - run: GOARCH=386 go test ./cmd/prometheus
```

Comment:
This seems to be doing 3 complete sets of tests plus one subset.
That feels like overkill to me; the point is to catch code that has been added which is incompatible with the older Go.  Do you think that further incompatibilities will be caught
under different tags or architecture, such that it's worth spending the time?

ISTM that `make test` would be simplest, but I could also argue that `--tags=stringlabels` is the mainline version Prometheus releases with.

Reply:
If we don't want to take any risks, I'd say yes, we should run the exact go tests than the current version.

- I'd even add the new `go test --tags=dedupelabels ./...`
- I can set  the new GO_ONLY=1 to `make test` to only run go tests.

If we're worried about resources, we can make it a daily test, but that may require more human resources...

Reply:
We can go with `make test` now, and enrich if we have a slip out in the future.

Reply:
The most precious resource is developer time.  If this test is the slowest part, then that's what everyone has to wait for: 
<img width="238" alt="image" src="https://github.com/prometheus/prometheus/assets/8125524/1e641ab1-61ba-4b54-9cdd-65c80c89781a">

I think 22 minutes is way overkill; please just run one set of tests.
"don't want to take any risks" is not something anyone said about Prometheus.

Reply:
Let's go with it then.
(Tell me if I should add the test with `-test.tsdb-isolation=false` back)